From 829ac0ad6f74f80c37867f76210c5c06cbf50605 Mon Sep 17 00:00:00 2001
From: Sam Bobrowicz <sam@elite-embedded.com>
Date: Wed, 26 Sep 2018 23:28:08 -0700
Subject: [PATCH 0469/1852] media: xilinx: csi2rxss: store format size

commit db798214773de5fc5b3654641e5bb61c213c6789 from
https://github.com/Xilinx/linux-xlnx.git

Currently set_format does not store the dimensions when setting a
non-RAW format. Although they are irrelevant to the operation of the
CSI RX IP, this can cause link_validate to fail and also unexpected
results when calling get_format. Fix this by storing the format size.

Signed-off-by: Sam Bobrowicz <sam@elite-embedded.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-csi2rxss.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-csi2rxss.c b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
index e38dbb061ac2..c2cca9b7e8db 100644
--- a/drivers/media/platform/xilinx/xilinx-csi2rxss.c
+++ b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
@@ -1327,6 +1327,8 @@ static int xcsi2rxss_set_format(struct v4l2_subdev *sd,
 		/* Restore the original pad format code */
 		fmt->format.code = code;
 		__format->code = code;
+		__format->width = fmt->format.width;
+		__format->height = fmt->format.height;
 	}
 
 	mutex_unlock(&xcsi2rxss->lock);
-- 
2.31.1

